palantir-java-format/build.gradle

```diff
@@ -32,6 +31,30 @@ dependencies {
 
 tasks.withType(JavaCompile).configureEach {
     options.errorprone.disable 'StrictUnusedVariable'
+
+    options.compilerArgs += [
+        // Allow access to internal javac apis
+        '--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED',
+        '--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED',
+        '--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED',
+        '--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED',
+        '--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED',
+        '--add-exports=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED',
+        '--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED'
+    ]
+
+    if (JavaVersion.current() < JavaVersion.VERSION_14) {
+        excludes = ['**/Java14InputAstVisitor.java']
```

Comment:
oooh nice, i was wondering how painful it was gonna be to do some kinda of dual compilation here, or if were going to have to do some multi-release jar thing!

One caveat with this approach is I think gradle has some new functionality where you can specify the JDK to use for these compile tasks independently from the JDK used to _evaluate_ gradle, and JavaVersion.current() will grab the gradle evaluating one